src/transformers/tokenization_utils_fast.py

```diff
@@ -112,6 +115,10 @@ def __init__(self, *args, **kwargs):
         elif slow_tokenizer is not None:
             # We need to convert a slow tokenizer to build the backend
             fast_tokenizer = convert_slow_tokenizer(slow_tokenizer)
+        elif from_gguf is not None:
+            # We need to convert a slow tokenizer to build the backend
+            tokenizer_dict = load_gguf_checkpoint(kwargs.get("vocab_file"))["tokenizer"]
```

Comment:
we read the metadata twice which is bad IMO. 
(another call is in tokenization auto to init the config)

Reply:
we only read the first bytes and we never read the entire file unless one passes `return_tensors=True` (it's set to `False` by default) so IMO it's not that bad !